specification/dartLangSpec.tex

```diff
@@ -23028,7 +23028,15 @@ \subsubsection{Comments}
 
 \LMHash{}%
 The current scope for a documentation comment immediately preceding
-the declaration of a function $f$ is the formal parameter scope of $f$.
+the declaration of a function $f$ is the
+\IndexCustom{formal parameter scope}{scope!formal parameter} of $f$.
+
+\LMHash{}%
+The current scope for a documentation comment immediately preceding
+the declaration of a non-redirecting generative constructor $c$
+with initializing formals is the
+\IndexCustom{formal parameter initializer scope}{%
+  scope!formal parameter initializer}of $c$.
```

Comment:
I'd recommend not creating an index entry for this page about 'formal parameter scope': This phrase is defined in the section 'Formal Parameters', and index entries in the language specification are created for the defining occurrence of the given phrase, not for usages. (We could have index entries for usages as well, but it would probably give rise to so many index entries that it would be unwieldy. Just search the document for usages using the 'Find' feature of the viewer.)

On the other hand, there are many references to the enclosing section when a phrase is used (that is, 'formal parameter scope' may be followed by a reference to the section 'Formal Parameters').

For some (unknown ;-) reason, constructors are usually called `$k$`.

So we could have something like this:

```suggestion
The current scope for a documentation comment immediately preceding
the declaration of a non-redirecting generative constructor $k$
with initializing formals is the formal parameter initializer scope of $k$
(\ref{generativeConstructors}).

\LMHash{}%
Otherwise, the current scope for a documentation comment immediately preceding
the declaration of a function $f$ is the formal parameter scope of $f$
(\ref{formalParameters}).
```